src/components/DrawModeSelector.tsx

```diff
@@ -0,0 +1,105 @@
+import React, { ReactElement, useRef } from "react";
+import styled from "styled-components";
+import LabeledDropdown from "./LabeledDropdown";
+import { DrawMode } from "../colorizer/ColorizeCanvas";
+import { Color as ThreeColor, ColorRepresentation } from "three";
+import { ColorPicker } from "antd";
+import { Color as AntdColor } from "@rc-component/color-picker";
+import { PresetsItem } from "antd/es/color-picker/interface";
+
+type DrawModeSelectorProps = {
+  label: string;
+  selected: DrawMode;
+  onChange: (mode: DrawMode, color: ThreeColor) => void;
+  color: ThreeColor;
+  disabled?: boolean;
+};
+
+const defaultProps: Partial<DrawModeSelectorProps> = {
+  disabled: false,
+};
+
+const MainLayout = styled.div`
+  display: flex;
+  flex-direction: column;
+  gap: 6px;
+`;
+
+const HorizontalDiv = styled(MainLayout)`
+  flex-direction: row;
+  flex-wrap: wrap;
+`;
+
+/**
+ * UI element for choosing between different drawing modes, and provides callbacks for when
+ * changes are made to selections.
+ * - `HIDE`: Hide an object type
+ * - `USE_COLOR`: Use a custom, solid color. (When selected, also shows a ColorPicker.)
+ * - `USE_RAMP`: Use the color ramp.
+ */
+export default function DrawModeSelector(propsInput: DrawModeSelectorProps): ReactElement {
```

Comment:
I think `Selector` is an overloaded term that is used in `Redux` to mean something totally different that is not a component.  I'll defer to others who know more about React/Redux as to how important it is to rename this. (@frasercl @meganrm )

Reply:
Library authors definitely like to use the generic word "selector" to mean lots of different non-user-facing things, but I don't think it would be disastrously confusing to see it as the name of a component. (I probably wouldn't ever expect a selector as Redux uses it to appear as a component.) If I were to rename the component I think I'd go with `DrawModeDropdown` to specify the generic term?

Reply:
There's also a ColorRampSelector currently that I could also rename (that is also a dropdown). Thank you both for the input!